### PR TITLE
add kfp v2 introduction

### PR DESCRIPTION
**Pull request by connor-mccarthy**
Some outstanding content, such as cross-section links, to be added once merged into the `pipelines-v2-docs` branch.

### CODE REVIEW

**Comment by connor-mccarthy** [comment]
/cc @chensun
for content review

File: content/en/docs/components/pipelines/introduction.md [hunk 1]
```diff
@@ -1,284 +1,60 @@
...
-  class="mt-3 mb-3 p-3 border border-info rounded">
-
-## Architectural overview
-
-<img src="/docs/images/pipelines-architecture.png" 
-  alt="Pipelines architectural diagram"
-  class="mt-3 mb-3 p-3 border border-info rounded">
-
-At a high level, the execution of a pipeline proceeds as follows: 
+A [_pipeline_][pipelines] is a description of an multi-step workflow, where each step is defined by a single container execution. Each step, or [_task_][tasks], is parameterized by inputs and outputs, enabling pipeline authors to form a computational directed acyclic graph (DAG) of tasks by specifying the output of one task as the input to another.
```
Comment (chensun) [inline content/en/docs/components/pipelines/introduction.md:30]: nit: a multi-step

(Can be single step as well)

File: content/en/docs/components/pipelines/introduction.md [hunk 2]
```diff
@@ -1,284 +1,60 @@
...
-    Kubeflow Pipelines stores the artifacts in an artifact store like
-    [Minio server](https://docs.minio.io/) or
-    [Cloud Storage](https://cloud.google.com/storage/docs/).
 
-    The MySQL database and the Minio server are both backed by the Kubernetes
-    [PersistentVolume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes)
-    subsystem.
+## What does using KFP look like?
+At a high level, a typical KFP user experience is as follows:
+1. [Author a pipeline][author-a-pipeline] using the **Python KFP SDK**'s domain-specific language (DSL)
```
Comment (chensun) [inline content/en/docs/components/pipelines/introduction.md:37]: A first step might be authoring a component, or reuse an prebuilt component. Not sure if we need to mention this though. Up to you.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**